csrc/id_model/id_model.cpp

```diff
@@ -204,6 +204,20 @@ void IdModel::buildIterDomainDefinitionsAndUses() {
         continue;
       }
 
+      // If any of the inputs is not included in the all ID set, do
```

Comment:
This is unrelated to the main change of this PR, but I just found a bug while creating the unit test. Basically, a definition expr of an ID should be ignored if its inputs are not included in `TensorView::allIDs()`.

Reply:
Does this actually cause problems in some cases through inappropriate merges? I am just wondering why it would be a problem to have a bit larger graph.

Reply:
Yes, without this change the added test fails at:

```
[ RUN      ] LoopDomainSchedulingTest.ScheduleLoopDomainsBy

Thread 1 "nvfuser_tests" hit Catchpoint 2 (exception thrown), 0x00007fffde7f835a in __cxa_throw () from /usr/lib/x86_64-linux-gnu/libstdc++.so.6
(gdb) bt
#0  0x00007fffde7f835a in __cxa_throw () from /usr/lib/x86_64-linux-gnu/libstdc++.so.6
#1  0x0000555556345753 in nvfuser::nvfCheckFail (func=0x555555eca72c <.L__FUNCTION__._ZNK7nvfuser12DisjointSetsIPNS_3ValESt4hashIS2_EE15strictAreMappedES2_S2_> "strictAreMapped",
    file=0x555555f3d22d <.L.str.98> "/home/nmaruyama/nvfuser/debug1/csrc/disjoint_set.h", line=420, msg=...) at /home/nmaruyama/nvfuser/debug1/csrc/exceptions.cpp:275
#2  0x00005555563459ad in nvfuser::nvfErrorFail (func=0x555555eca72c <.L__FUNCTION__._ZNK7nvfuser12DisjointSetsIPNS_3ValESt4hashIS2_EE15strictAreMappedES2_S2_> "strictAreMapped",
    file=0x555555f3d22d <.L.str.98> "/home/nmaruyama/nvfuser/debug1/csrc/disjoint_set.h", line=420,
    condMsg=0x555555e7426a <.L.str.90> " INTERNAL ASSERT FAILED at \"/home/nmaruyama/nvfuser/debug1/csrc/disjoint_set.h\":420, please report a bug with repro script to NVFuser at https://github.com/NVIDIA/Fuser/issues. ", userMsg=...)
    at /home/nmaruyama/nvfuser/debug1/csrc/exceptions.cpp:301
#3  0x00005555560b82c1 in nvfuser::DisjointSets<nvfuser::Val*, std::hash<nvfuser::Val*> >::strictAreMapped (this=0x555557bbc208, entry0=0x55555af961f0, entry1=0x55555af91ae0) at /home/nmaruyama/nvfuser/debug1/csrc/disjoint_set.h:416
#4  0x0000555556a2f621 in nvfuser::ValGraph::mapVals (this=0x555557bbc200, val0=0x55555af961f0, val1=0x55555af91ae0) at /home/nmaruyama/nvfuser/debug1/csrc/val_graph.cpp:401
#5  0x0000555556a30fb1 in nvfuser::ValGraph::mapThroughExpr (this=0x555557bbc200, first=0x55555af96350, second=0x55555af923e0, forward=false) at /home/nmaruyama/nvfuser/debug1/csrc/val_graph.cpp:560
#6  0x0000555556a301eb in nvfuser::ValGraph::maybeMapThroughExprs (this=0x555557bbc200, expr0=0x55555af96350, expr1=0x55555af923e0, forward=false) at /home/nmaruyama/nvfuser/debug1/csrc/val_graph.cpp:490
#7  0x0000555556a30001 in nvfuser::ValGraph::mapVals (this=0x555557bbc200, val0=0x55555af93570, val1=0x55555af92260) at /home/nmaruyama/nvfuser/debug1/csrc/val_graph.cpp:459
#8  0x0000555556484ec7 in nvfuser::IdModel::buildExactGraph (this=0x7fffffffcc90) at /home/nmaruyama/nvfuser/debug1/csrc/id_model/id_model.cpp:317
#9  0x0000555556967ebf in nvfuser::scheduler_tools::scheduleLoopDomainsBy (tvs=..., transform=0x55555af923e0) at /home/nmaruyama/nvfuser/debug1/csrc/scheduler/tools/loop_domain_scheduler.cpp:393
#10 0x000055555729cddd in nvfuser::LoopDomainSchedulingTest_ScheduleLoopDomainsBy_Test::TestBody (this=0x55555af85fe0) at /home/nmaruyama/nvfuser/debug1/tests/cpp/test_loop_domain_scheduling.cpp:377
```

I believe this is because when an exact graph is built, mapping is propagated through exprs, which means if the outputs and the expr themselves are mapped, respectively, their inputs are going to be mapped, but if the inputs are not registered in the ValGraph itself, `strictAreMapped` fails as shown above.